pkg/analysis/optionalfields/util.go

```diff
@@ -0,0 +1,273 @@
+// instace of the marker with the given name.
+// Works for markers like MaxLength, MinLength, etc.
+func getMarkerIntegerValueByName(marker markers.MarkerSet, markerName string) (*int, error) {
+	markerList := marker.Get(markerName)
+	if len(markerList) == 0 {
+		return nil, errMarkerMissingValue
+	}
+
+	markerValue, err := getMarkerIntegerValue(markerList[0])
+	if err != nil {
+		return nil, fmt.Errorf("error getting marker value: %w", err)
+	}
+
+	return &markerValue, nil
+}
+
+// getMarkerIntegerValue extracts a numeric value from the default
+// value of a marker.
+// Works for markers like MaxLength, MinLength, etc.
+func getMarkerIntegerValue(marker markers.Marker) (int, error) {
+	rawValue, ok := marker.Expressions[""]
+	if !ok {
+		return 0, errMarkerMissingValue
+	}
+
+	value, err := strconv.Atoi(rawValue)
+	if err != nil {
+		return 0, fmt.Errorf("error converting value to integer: %w", err)
+	}
+
+	return value, nil
+}
+
+// getMarkerFloatValueByName extracts the numeric value from the first
+// instace of the marker with the given name.
+// Works for markers like MaxLength, MinLength, etc.
+func getMarkerFloatValueByName(marker markers.MarkerSet, markerName string) (*float64, error) {
+	markerList := marker.Get(markerName)
+	if len(markerList) == 0 {
+		return nil, errMarkerMissingValue
+	}
+
+	markerValue, err := getMarkerFloatValue(markerList[0])
+	if err != nil {
+		return nil, fmt.Errorf("error getting marker value: %w", err)
+	}
+
+	return &markerValue, nil
+}
+
+// getMarkerFloatValue extracts a numeric value from the default
+// value of a marker.
+// Works for markers like MaxLength, MinLength, etc.
+func getMarkerFloatValue(marker markers.Marker) (float64, error) {
+	rawValue, ok := marker.Expressions[""]
+	if !ok {
+		return 0, errMarkerMissingValue
+	}
+
+	value, err := strconv.ParseFloat(rawValue, 64)
+	if err != nil {
+		return 0, fmt.Errorf("error converting value to float: %w", err)
+	}
+
+	return value, nil
+}
+
+// structHasGreaterThanZeroMaxProperties checks if the struct has a minProperties marker.
+// It inspects the minProperties marker on the struct type itself.
+func structHasGreaterThanZeroMinProperties(structType *ast.StructType, structMarkers markers.MarkerSet) (bool, error) {
+	if structType == nil {
+		return false, nil
+	}
+
+	for _, marker := range structMarkers.Get(minPropertiesMarker) {
+		markerValue, err := getMarkerIntegerValue(marker)
+		if err != nil {
+			return false, fmt.Errorf("error getting marker value: %w", err)
+		}
+
+		if markerValue > 0 {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func integerRangeIncludesZero(minimum, maximum *int) bool {
+	return ptr.Deref(minimum, -1) == 0 ||
+		ptr.Deref(maximum, -1) == 0 ||
+		ptr.Deref(minimum, 0) < 0 && ptr.Deref(maximum, 0) > 0
+}
+
+func floatRangeIncludesZero(minimum, maximum *float64) bool {
+	return ptr.Deref(minimum, -1) == 0 ||
+		ptr.Deref(maximum, -1) == 0 ||
+		ptr.Deref(minimum, 0) < 0 && ptr.Deref(maximum, 0) > 0
+}
```

Comment:
Nit: This _could_ use generics since this is a pretty straightforward "does range of min/max contain 0"